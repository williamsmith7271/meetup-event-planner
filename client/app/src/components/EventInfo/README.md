## EventInfo Component
A component that takes an event and renders the event's info

### Example

```js
<EventInfo event={event} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **event**    | Object   |             | An object, with the shape shown below

```
event: PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  end: PropTypes.string.isRequired,
  start: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  host: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  guests: PropTypes.array.isRequired,
}),
```
