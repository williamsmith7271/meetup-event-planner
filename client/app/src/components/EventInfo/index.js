import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import Paragraph from 'grommet-udacity/components/Paragraph';
import Button from 'grommet-udacity/components/Button';
import Footer from 'grommet-udacity/components/Footer';
import ScheduleNewIcon from 'grommet-udacity/components/icons/base/ScheduleNew';
import moment from 'moment';

const formatString = 'MMMM Do YYYY, h:mm:ss a';
const parseDates = (a, b) =>
  `From ${moment(a).format(formatString)} to ${moment(b).format(formatString)}`;

const EventInfo = ({
  event,
}) => (
  <Box
    colorIndex="light-1"
    size="auto"
    pad={{ horizontal: 'large', vertical: 'medium' }}
    className={styles.eventInfo}
  >
    <Heading className={styles.eventHeading} align="start" tag="h2">
      {event.name}
    </Heading>
    <Heading align="end" tag="h4">
      {event.type.charAt(0).toUpperCase() + event.type.slice(1)}
    </Heading>
    <Heading align="end" tag="h5">
      Hosted by {event.host.name}
    </Heading>
    <Heading align="end" tag="h5">
      {parseDates(event.startDate, event.endDate)}
    </Heading>
    <Heading align="end" tag="h5">
      {`${event.guests.length} people are attending`}
    </Heading>
    <Paragraph>
      {event.message}
    </Paragraph>
    <Footer justify="center">
      <Button
        href={`/events/${event.id}`}
        label="View Details"
        icon={<ScheduleNewIcon />}
        onClick={e => e}
      />
    </Footer>
  </Box>
);

EventInfo.propTypes = {
  event: PropTypes.object.isRequired,
};

export default cssModules(EventInfo, styles);
