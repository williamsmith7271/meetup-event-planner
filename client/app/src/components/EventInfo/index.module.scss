.eventInfo {
  border-radius: 5px;
  border: 1px solid #e6e6e6;
  margin-bottom: 30px;
  padding: 30px !important;
  border-radius: 5px;
  @media screen and (min-width: 550px) {
    min-width: 580px;
    max-width: 100vw;
  }
}

.eventHeading {
  border-bottom: 1px solid rgba(0,0,0,.12);
}
