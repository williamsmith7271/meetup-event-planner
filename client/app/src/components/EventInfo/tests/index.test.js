import EventInfo from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<EventInfo />', () => {
  it('should render with default props', () => {
    const event = {
      id: '4',
      name: 'Pool Party',
      end: '2014-12-12 00:00:00 UTC',
      start: '2016-11-02 00:00:00 UTC',
      type: 'birthday',
      location: '1204 North Carolina 55, Fuquay Varina, NC, United States',
      message: 'The best party',
      host: { name: 'Joaquin West' },
      guests: [{ name: 'Bilbo Baggins' }],
    };
    const wrapper = shallow(
      <EventInfo event={event} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
