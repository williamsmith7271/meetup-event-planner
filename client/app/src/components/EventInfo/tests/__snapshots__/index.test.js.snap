exports[`<EventInfo /> should render with default props 1`] = `
<Box
  announce={false}
  colorIndex="light-1"
  direction="column"
  focusable={true}
  pad={
    Object {
      "horizontal": "large",
      "vertical": "medium"
    }
  }
  responsive={true}
  size="auto"
  tag="div">
  <Heading
    align="start"
    tag="h2">
    Pool Party
  </Heading>
  <Heading
    align="end"
    tag="h4">
    Birthday
  </Heading>
  <Heading
    align="end"
    tag="h5">
    Hosted by 
    Joaquin West
  </Heading>
  <Heading
    align="end"
    tag="h5">
    From October 12th 2016, 3:07:35 pm to October 12th 2016, 3:07:35 pm
  </Heading>
  <Heading
    align="end"
    tag="h5">
    1 people are attending
  </Heading>
  <Paragraph>
    The best party
  </Paragraph>
  <Footer
    direction="row"
    justify="center"
    responsive={false}>
    <Button
      href="/events/4"
      icon={
        <function Icon() {
            (0, _classCallCheck3.default)(this, Icon);
            return (0, _possibleConstructorReturn3.default)(this, (Icon.__proto__ || (0, _getPrototypeOf2.default)(Icon)).apply(this, arguments));
          }
          a11yTitleId="schedule-new-title"
          responsive={true} />
      }
      label="View Details"
      onClick={[Function onClick]}
      type="button" />
  </Footer>
</Box>
`;
